usr/share/rear/lib/network-functions.sh

```diff
@@ -184,7 +184,11 @@ function is_persistent_ethernet_name () {
     local _name_assign_type="0"
 
     [ -f "/sys/class/net/$_netif/name_assign_type" ] \
-        && _name_assign_type=$(cat "/sys/class/net/$_netif/name_assign_type")
+        && _name_assign_type=$(cat "/sys/class/net/$_netif/name_assign_type" 2>/dev/null)
+
+    # On RHEL 6 "name_assign_type" does not exist, therefore, the value of _name_assign_type=""
+    # Read https://github.com/rear/rear/issues/2197 for the details
+    [ "$_name_assign_type" = "" ] && return 1
```

Comment:
Now, if `/sys/class/net/$_netif/name_assign_type` does not exist, the value of _name_assign_type is 0 and the behaviour is different than if it exists, but can not be read. Is that intended? If the file is missing, should not the result be the same as if it is unreadable?

Reply:
@pcahyna On RHEL 6 `/sys/class/net/$_netif/name_assign_type` does not exist, therefore, the value is _empty_ and not 0. IMHO the return code must be 1 if the file is missing (or even when it is unreadable).

Reply:
It does exist, but it is unreadable, that's what #2197 is about. If it did not exist, the 
```
[ -f "/sys/class/net/$_netif/name_assign_type" ] \
&&
```
would protect against reading it, so non-existence would lead to the value being 0, not "".
I actually tried on RHEL 6:
```
# uname -r
2.6.32-754.el6.x86_64
# ls -l /sys/class/net/eth0/name_assign_type 
-r--r--r-- 1 root root 4096 Sep 10 05:48 /sys/class/net/eth0/name_assign_type
# cat /sys/class/net/eth0/name_assign_type 
cat: /sys/class/net/eth0/name_assign_type: Invalid argument
```
So, the return value is not necessarily 1 when the file is missing (but after your fix, it is 1 when it is present and unreadable. This is the inconsistency I am pointing out.)

Reply:
@pcahyna right I can follow you. The cosmetic change makes no sense in this case as the following case statement will make the right decision anyhow. Will withdraw the PR